src/cloud/audiocom/ShareAudioDialog.cpp

```diff
@@ -0,0 +1,842 @@
+   {
+      return FileExtension { "WAV" };
+   }
+
+   FileExtension GetExtension() const override
+   {
+      return FileExtension { "wav" };
+   }
+};
+
+#ifdef USE_LIBFLAC
+struct FlacExportHelper final : ExportHelper
+{
+   explicit FlacExportHelper(AudacityProject& project)
+       : ExportHelper(project)
+   {
+   }
+
+   void SetupFormat() override
+   {
+      FLACBitDepth.Write("24");
+      FLACLevel.Write("5");
+   }
+
+   FileExtension GetFormat() const override
+   {
+      return FileExtension { "FLAC" };
+   }
+
+   FileExtension GetExtension() const override
+   {
+      return FileExtension { "flac" };
+   }
+};
+#endif
+
+#ifdef USE_WAVPACK
+struct WavPackExportHelper final : ExportHelper
+{
+   explicit WavPackExportHelper(AudacityProject& project)
+       : ExportHelper(project)
+   {
+   }
+
+   void SetupFormat() override
+   {
+      QualitySetting.Write(2);
+      BitrateSetting.Write(40);
+      BitDepthSetting.Write(24);
+      HybridModeSetting.Write(false);
+   }
+
+   FileExtension GetFormat() const override
+   {
+      return FileExtension { "WavPack" };
+   }
+
+   FileExtension GetExtension() const override
+   {
+      return FileExtension { "wv" };
+   }
+};
+#endif
+
+// We use WAV only as a fall back (unless config asks for it explicitly)
+std::unique_ptr<ExportHelper>
+GetExportHelper(AudacityProject& project)
+{
+   const auto preferredFormat = GetServiceConfig().GetPreferredAudioFormat();
+
+#ifdef USE_WAVPACK
+   if (preferredFormat == AudioFormat::WAVPACK)
+      return std::make_unique<WavPackExportHelper>(project);
+#endif
+#ifdef USE_LIBFLAC
+   if (preferredFormat != AudioFormat::WAV)
+      return std::make_unique<FlacExportHelper>(project);
+#endif
+   return std::make_unique<WavExportHelper>(project);
+}
+}
+
+// A helper structures holds UploadService and UploadPromise
+struct ShareAudioDialog::Services final
+{
+   UploadService uploadService;
+
+   std::shared_ptr<UploadPromise> uploadPromise;
+
+   Services()
+       : uploadService(GetServiceConfig(), GetOAuthService())
+   {
+   }
+};
+
+// Implementation of the ProgressDialog, which is not a dialog.
+// Instead, progress is forwarded to the parent
+struct ShareAudioDialog::ExportProgressHelper final :
+    public BasicUI::ProgressDialog
+{
+   explicit ExportProgressHelper(ShareAudioDialog& parent)
+       : mParent(parent)
+   {
+   }
+
+   void Cancel()
+   {
+      mCancelled = true;
+   }
+
+   bool WasCancelled()
+   {
+      return mCancelled;
+   }
+
+   BasicUI::ProgressResult Poll(
+      unsigned long long numerator, unsigned long long denominator,
+      const TranslatableString&) override
+   {
+      mParent.UpdateProgress(numerator, denominator);
+
+      const auto now = Clock::now();
+
+      // Exporter polls in the main thread. To make the dialog responsive
+      // periodic yielding is required
+      if ((now - mLastYield > std::chrono::milliseconds(50)) || (numerator == denominator))
```

Comment:
You want a busy-wait when numerator equals denominator?

Reply:
This is the behavior from the original progress bar.